script.js

```diff
@@ -0,0 +1,942 @@
+      addRemoveKeyActive(key);
+      target = key;
+      break;
+    } else if (isKeydown && !e.repeat
+      && (keyText === variables.specialKeys.CAPS_LOCK && e.key === 'CapsLock')) {
+      if (key.classList.contains(classes.KEY_ACTIVE)) {
+        key.classList.remove(classes.KEY_ACTIVE);
+      } else {
+        key.classList.add(classes.KEY_ACTIVE);
+      }
+      target = key;
+      break;
+    } else if (isKeydown && !e.repeat
+      && (keyText === variables.specialKeys.NUM_LOCK && e.key === 'NumLock')) {
+      if (key.classList.contains(classes.KEY_ACTIVE)) {
+        key.classList.remove(classes.KEY_ACTIVE);
+      } else {
+        key.classList.add(classes.KEY_ACTIVE);
+      }
+      target = key;
+      break;
+    } else if (isSecondKey(keyText, e)) {
+      if (isRightKey) {
+        addRemoveKeyActive(key);
+        target = key;
+        break;
+      }
+      isRightKey = true;
+    } else if (keyText === e.key) {
+      addRemoveKeyActive(key);
+      target = key;
+      break;
+    }
+  }
+
+  return target;
+};
+
+// ///////////////////////// KEYBOARD HANDLERS ///////////////////////////
+const handlerKeyDown = (e) => {
+  if (!e.repeat) {
+    playKeypressSound();
+
+    if (((e.ctrlKey && e.shiftKey) && (e.code === 'ControlLeft' || e.code === 'ShiftLeft'))
+    || ((e.ctrlKey && e.altKey) && (e.code === 'ControlLeft' || e.code === 'AltLeft'))
+    || ((e.shiftKey && e.altKey) && (e.code === 'ShiftLeft' || e.code === 'AltLeft'))) {
+      changeLanguage();
+    }
+  }
+
+  elements.textarea.focus();
+
+  const target = processKeySelection(e);
+  handlerKeyInput(target, e);
+};
+
+const handlerKeyUp = (e) => {
+  if (e.key === variables.specialKeys.SHIFT) {
+    variables.isShift = !variables.isShift;
+    changeCase();
+    changeOnShift();
+  }
+
+  const PRINT_SCREEN = 'PrintScreen';
+  if (e.key === PRINT_SCREEN) {
+    const customEvent = new KeyboardEvent('keydown', {
+      cancelable: true,
+      key: PRINT_SCREEN,
+      code: PRINT_SCREEN,
+      ctrlKey: false,
+      shiftKey: false,
+      altKey: false,
+      repeat: false,
+    });
+    handlerKeyDown(customEvent);
+    setTimeout(() => processKeySelection(e), 100);
+    return;
+  }
+  // если отпустили - сбросить зажатие клавиши
+  processKeySelection(e);
+};
+
+// ///////////////////////// MOUSE HANDLERS ///////////////////////////
+const handlerMouseDown = (e) => {
+  // find div.key
+  let target;
+  if (e.target.classList.contains(classes.KEY)) {
+    target = e.target;
+  } else if (!e.target.children.length) {
+    target = e.target.parentElement;
+  }
+  // store this node as a fired mousedown event to the future removal
+  variables.mousedownFiredEvent = target;
+
+  // make UI effects if event exists
+  if (target) {
+    playKeypressSound();
+
+    const text = target.firstElementChild.innerText;
+    if ((e.ctrlKey && text === variables.specialKeys.SHIFT)
+    || (e.ctrlKey && text === variables.specialKeys.ALT)
```

Comment:
Very hard to read this if statement

Reply:
Try to clean it. Create some meaningful functions like this.
![image](https://user-images.githubusercontent.com/29872339/79618651-53e37a00-8113-11ea-9c13-d6995a81c0e9.png)

Reply:
I don't think, that it is useful for me, because I wanted to check, if text not present in the object, but if text is equal to SHIFT, or ALT. Different logics here :)

У меня не выходит придумать нечто более логичное, чем литералы строк объединённые в своеобразные константы-поля объекта

Reply:
Fixed, but in another way:
```js
const isShift = (key) => key === variables.specialKeys.SHIFT;
    const isCtrl = (key) => key === variables.specialKeys.CTRL;
    const isAlt = (key) => key === variables.specialKeys.ALT;
    const isCtrlShiftPressed = (e.ctrlKey && isShift(text)) || (e.shiftKey && isCtrl(text));
    const isCtrlAltPressed = (e.ctrlKey && isAlt(text)) || (e.altKey && isCtrl(text));
    const isAltShiftPressed = (e.altKey && isShift(text)) || (e.shiftKey && isAlt(text));

    if (isCtrlShiftPressed || isCtrlAltPressed || isAltShiftPressed) {
      changeLanguage();
    }
```